### DISCUSSION TITLE
Erode all to 1 pixel, is this possible?

### DISCUSSION THREAD

**Discussion by LDAsh72** [ImageMagick/ImageMagick#2637] [Help] [answered]
Any version of IM will do, infact any other software would even do.
I'm trying to take a layout of shapes and erode them all down to 1 pixel.  These shapes may vary in size, but I need them to retain that 1 central pixel.  Like in this example:-
http://www.violae.net/temp/erode21.png

**Comment by snibgo** [comment, marked as answer]
Are they white marks always square? Do you care which point in the square is found? This finds the top-left corner of each square:
```
magick erode21.png -morphology HMT 2x2:0,0,0,1 x.png
```

**Reply by LDAsh72** [reply]
Thanks, that's great enough!  I wondered by it needed to be 2x2 pixels, but the easy fix is just to make the input 3X larger and then sample down after the process, to get my 1 pixels.  A simple "roll" will offset everything to be more or less in the middle, so yeah - good times!

**Comment by fmw42** [comment]
You can use -connected-components to find the centroids of each white object. Then draw a point or rectangle for each on a black background.
